### PR TITLE
[CI] Reduce the number of test pipelines in CI

### PR DESCRIPTION
**Pull request by dpakach**
## Description

Reduce the number of pipelines running in the CI by merging the suites.
Also add code to check if any test suites are missing in the CI run.

Number of pipelines running before this PR - 100
Number of pipelines running after this PR - 64

When we run multiple pipelines, it needs to setup server, run databases, selenium etc on each pipeline so reducing this number will hopefully reduce the load in CI server

## Related Issue

- Fixes https://github.com/owncloud/web/issues/5158

## How Has This Been Tested?

#:robot: 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] ...

### CODE REVIEW

File: .drone.star
Comment (phil-davis) [inline .drone.star:102]: Did `webUIUpload` get lost?

Comment (dpakach) [reply .drone.star:102]: Its not lost 
https://github.com/owncloud/web/pull/5195/files#diff-6a968d4b7901c8526c8d9a86d1f434737dbfe2b00da21a7d3041ed4e3d58d62eR56

But this approache is prone to accidently losing an entire suite in that way, so I'm trying to make a function that checks all the available test suites and see if any suites are missing in the tests.

Comment (phil-davis) [reply .drone.star:102]: OK - it moved a long way and I lost track of it.

**Review by phil-davis — APPROVED** [review]
The changes look good. Please write some text and link to the issue before merging.

**Comment by dpakach** [comment]
@phil-davis I've added the code to check if any test suites are missing, please review and comment what you think of this approach?

**Review by individual-it — APPROVED** [review]

**Review by individual-it — COMMENTED** [review]
please only merge after @kulmann or @LukasHirt approved, so that it does not disturb their process in  #5018

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingPublicManagement</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/15958/

<p>


### webUISharingPublicManagement-shareByPublicLink-feature-90.png
  


</p></details>
<details><summary>:boom: Acceptance tests <strong>SharingPublicManagement</strong> failed. The build is cancelled...</summary>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingPublicManagement</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/15961/

<p>


### webUISharingPublicManagement-shareByPublicLink-feature-90.png
  


</p></details>
<details><summary>:boom: Acceptance tests <strong>SharingPublicManagement</strong> failed. The build is cancelled...</summary>

File: .drone.star
Comment (phil-davis) [inline .drone.star:604]:
```suggestion
notificationsTestSuites = [
```
POLA - I am astonished to see multiple items in this array, but no "s" at the end of the name ;)

**Review by phil-davis — COMMENTED** [review]
Just one tiny name thing.

It would be nice if there was a way to get a list of the directories in `tests/acceptance/features` but the starlark needs to run before the pipelines are given to the agents, before there is a clone of the repo available.

Duplicating the lists in a different and cross-checking will at least pickup errors when people just shuffle the suite lists in the config section. It won't automatically notice if there are new directories (suites) of features added to the repo.

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingPublicManagement</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/15971/

<p>


### webUISharingPublicManagement-shareByPublicLink-feature-90.png
  


</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUIBasic</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/15971/

<p>


### webUILogin-adminBlocksUser-feature-20.png
  

### webUIPreview-imageMediaViewer-feature-140.png
  

### webUIPreview-imageMediaViewer-feature-158.png
  

### webUIPreview-imageMediaViewer-feature-83.png
  

### webUIPreview-imageMediaViewer-feature-90.png
  

### webUIPrivateLinks-accessingPrivateLinks-feature-17.png
  

### webUIPrivateLinks-accessingPrivateLinks-feature-25.png
  

### webUIPrivateLinks-accessingPrivateLinks-feature-34.png
  

### webUIPrivateLinks-accessingPrivateLinks-feature-9.png
  


</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>IntegrationApp1</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/15971/

<p>


### webUIPrivateLinks-accessingPrivateLinks-feature-17.png
  

### webUIPrivateLinks-accessingPrivateLinks-feature-9.png
  


</p></details>

File: .drone.star
Comment (phil-davis) [reply .drone.star]: Done

**Review by phil-davis — APPROVED** [review]

**Comment by phil-davis** [comment]
Can be merged when CI finishes.

**Comment by dpakach** [comment]
The CI failed in npm install because it couldn't pull the docker image, i've restarted it..

**Pull request merged**
